[CI] add decorator for specifying H100-only tests

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #156980

Motivation: test_aot_inductor and test_triton_kernels have a small fraction of their tests which need to be run on sm90. I want to run those tests on the H100 smoke test CI job, but adding the entire test suites for test_aot_inductor and test_triton_kernels seems like a waste of H100 time. 

To solve this: this PR adds `@filterByNvidiaArch()` - a decorator that marks a class as potentially containing H100-specific tests - and adds `@runOnNvidiaArch(arch_list)` - a decorator that marks a test as one that should run on H100 hardware.

`@filterByNvidiaArch` will check for the `TORCH_TEST_FILTER_NVIDIA_ARCH` env variable - and if it is set, it will remove any tests that don't have the `@runOnNvidiaArch`  decorator for the appropriate arch.

This PR also adds test_aot_inductor and test_triton_kernels to the H100 CI job and marks the relevant tests to be run on H100

RFC: Rocm already has runOnRocmArch, which has different semantics; if it's important to fix this, LMK and I can try unifying them if you want, or renaming this one something different, etc.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

davidberard98 committed:
[CI] add decorator for specifying H100-only tests

[ghstack-poisoned]

davidberard98 commented:
@eqy 

IIUC SM90OrLater is just a boolean and not a decorator, and it's used as `@unittest.skipIf(not SM90OrLater, "Grouped gemm supported on SM90")`, and it means "skip this test if we're not on SM90 or later".

And what I want is "skip **all tests except the ones marked with @filterByNvidiaArch('sm90')**".

So I don't think SM90OrLater is sufficient. But I'm definitely open to any suggestions for different ways to express this idea if you have any!

eqy commented:
ah understood, I wonder from this perspective we could bake the compute capability specific nature in the test name and rely on existing unittest/pytest filtering?

no objections to current approach

davidberard98 committed:
Update

[ghstack-poisoned]

davidberard98 committed:
Update on "[CI] add decorator for specifying H100-only tests"


Motivation: test_aot_inductor and test_triton_kernels have a small fraction of their tests which need to be run on sm90. I want to run those tests on the H100 smoke test CI job, but adding the entire test suites for test_aot_inductor and test_triton_kernels seems like a waste of H100 time. 

To solve this: this PR adds `filterByNvidiaArch()` - a decorator that marks a class as potentially containing H100-specific tests - and adds `runOnNvidiaArch(arch_list)` - a decorator that marks a test as one that should run on H100 hardware.

`filterByNvidiaArch` will check for the `TORCH_TEST_FILTER_NVIDIA_ARCH` env variable - and if it is set, it will remove any tests that don't have the `runOnNvidiaArch`  decorator for the appropriate arch.

This PR also adds test_aot_inductor and test_triton_kernels to the H100 CI job and marks the relevant tests to be run on H100

RFC: Rocm already has runOnRocmArch, which has different semantics; if it's important to fix this, LMK and I can try unifying them if you want, or renaming this one something different, etc.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
